### ISSUE TITLE
Mistake in sequential version number check

### ISSUE THREAD

**Issue by fredrikekre** [JuliaRegistries/RegistryCI.jl#49]
https://github.com/JuliaRegistries/General/pull/4722

**Comment by DilumAluthge** [comment]
So, this one is really tricky. In this case, the latest registered version was 0.9.1.

In this particular case, the author intentionally wanted to register version 0.7.5.

But in other cases, authors have accidentally created non-sequential releases when they did not intend to.

So I think we need to have some mechanism for an author to tell automerge that they intentionally wanted to create a non-sequential release. Basically, they need to give automerge the following information:
1. Is this a normal sequential release? `Yes` or `no`. The default value is `yes`.
2. If the answer to number 1 is no, what is the relevant release line. In this case, the value would be `0.7`.

If the answer to number 1 is yes, then we perform the check that we current do - we pull the latest version number and make sure that the increment from the latest version number to the proposed new version is either a.b.c -> a.b.(c+1), a.b.c -> a.(b+1).0, or a.b.c -> (a+1).0.0.

If the answer to number 1 is no, then we retrieve the answer to number 2, and then we pull the latest release in the appropriate release line. For example, if the provided answer to number 2 is `0.7`, then for this example, the latest release in the `0.7` line was `0.7.4`. So we pull the version number `0.7.4`. Then we make sure that the increment from this version number `0.7.4` to the proposed new version `0.7.5` matches a.b.c -> a.b.(c+1), a.b.c -> a.(b+1).0, or a.b.c -> (a+1).0.0.

But I think that this should be opt in behavior.

So if a user wants to make a release like this, I think there should be a way for them to provide the answers to number 1 and number 2 to Registrator.

Then, Registrator puts this information in the title of the pull request.

Then automerge retrieves the information from the title of the pull request.

**Comment by DilumAluthge** [comment]
TL;DR: I think we should add support for automerging this kind of nonsequential release. But I think it should be opt-in, and I think we need to first add a feature to Registrator to collect this information.

**Comment by StefanKarpinski** [comment]
It should be fine to register `0.7.5` as long as `0.7.4` exists and is registered. There's a whole tree of viable new releases at any point in time, not just one. In this circumstance, `0.7.5`, `0.9.2` and `0.10.0` should all be considered unobjectionable new release versions.

**Comment by StefanKarpinski** [comment]
There used to be a utility function in Base that checked if a new version number was acceptable based on a set of existing version numbers. It was removed in https://github.com/JuliaLang/julia/commit/5741c7f53c5ea443bbd73cc5157b83de415c6d1b.

**Comment by DilumAluthge** [comment]
Ah I see. Alright should not be too difficult to fix this.

**Comment by DilumAluthge** [comment]
This is an important bug to fix. It would be great if someone could help out.

**Comment by ericphanson** [comment]
Should we allow pre-release and build annotations? e.g. `v"2-rc"`?

**Comment by DilumAluthge** [comment]
We don’t allow that.
